src/Extensions/Hosting.Services.Web/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Fabric;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.ServiceFabric.Services.Communication.AspNetCore;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web
+{
+	/// <summary>
+	/// Extension to add Omex dependencies to HostBuilder
+	/// </summary>
+	public static class HostBuilderExtensions
+	{
+		/// <summary>
+		/// Add Kestrel service listener to stateless service
+		/// </summary>
+		public static ServiceFabricHostBuilder<StatelessServiceContext> AddKestrelListener<TStartup>(
+			this ServiceFabricHostBuilder<StatelessServiceContext> builder,
+			string name,
+			ServiceFabricIntegrationOptions options,
+			Action<IWebHostBuilder>? builderExtension = null)
+				where TStartup : class =>
+			builder.AddKestrelListener<TStartup, StatelessServiceContext>(name, options, builderExtension);
+
+
+		/// <summary>
+		/// Add Kestrel service listener to stateful service
+		/// </summary>
+		public static ServiceFabricHostBuilder<StatefulServiceContext> AddKestrelListener<TStartup>(
+			this ServiceFabricHostBuilder<StatefulServiceContext> builder,
```

Comment:
I think you can combine this with the method above by making `StatefulServiceContext` a typeparam

Reply:
Indeed I can, that's the way it was done initially, but then users would need to specify both type parameters (Context type and startup class) when calling the method. Since the context type always could be inferred from usage and typing it would be wasteful I've decided to write two methods for better user experience.